### PR TITLE
Upgrade Elasticsearch image to 6.5.4

### PR DESCRIPTION
**Pull request by jraddaoui**
- Declare node name and other configurations in environment section.
- Remove `privileged` and `expose` parameters.

Needed after https://github.com/artefactual/archivematica/pull/1285 gets merged and its submodule updated in here.

Connects to https://github.com/artefactual/archivematica/issues/1171.

### CODE REVIEW

**Comment by jraddaoui** [comment]
Hi @sevein,

I think there is no need to use `expose` when the port is included in `ports`, so we could remove that from all the services. But I wonder why was the `privileged` option in the ES service, is that needed using a named volume?

**Comment by sevein** [comment]
> I think there is no need to use expose when the port is included in ports, so we could remove that from all the services.

I didn't know that, cool. So from other containers, the original port would remain right? E.g. 9200 from a container, 62002 from the host?

> But I wonder why was the privileged option in the ES service, is that needed using a named volume?

**Comment by jraddaoui** [comment]
> I didn't know that, cool. So from other containers, the original port would remain right? E.g. 9200 from a container, 62002 from the host?

That's right ;)

File: compose/docker-compose.yml [hunk 1]
```diff
@@ -38,13 +38,19 @@ services:
...
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
     volumes:
       - "elasticsearch_data:/usr/share/elasticsearch/data"
-    expose:
-      - "9200"
```
Comment (sevein) [inline compose/docker-compose.yml]: Would you be able to make the same change in the other services for consistency?

Comment (jraddaoui) [reply compose/docker-compose.yml]: Will do ;)

**Comment by jraddaoui** [comment]
I'll update the archivematica sub-module when artefactual/archivematica#1171 is merged and merge this PR.

**Comment by sevein** [comment]
@jraddaoui will the `flush-elasticsearch-indices` rule need to be updated?

https://github.com/artefactual-labs/am/blob/f8596df7e1a4a5ea73728b4c76ab6d1552a9a2e7/compose/Makefile#L133-L134

**Comment by jraddaoui** [comment]
Yes, good point!

**Comment by jraddaoui** [comment]
I've fixed the flush ES indexes rule and changed the image again to a more recent 6.x version. I think it's okay to merge this PR with `--ff-only` once I update the sub-modules.

**Comment by ross-spencer** [comment]
@sevein @jraddaoui would it be prudent to cut a stable/1.8.x branch for am.git before merging into master? I suspect the additional indices will break the Makefile for 1.8. It might be good to have one for 1.8 anyway, but what do y'all think?

**Comment by sevein** [comment]
> @sevein @jraddaoui would it be prudent to cut a stable/1.8.x branch for am.git before merging into master?

That's a good idea. But there would be more work needed, e.g. tweak the project name, submodules, volumes, etc... I'd say that it almost deserves a new ticket maybe? I would personally be happy not making it a priority nor a blocker since we haven't really made a promise about what's supported other than the latest development tip of the project. Or maybe we did?

**Comment by ross-spencer** [comment]
Definitely its own ticket. Might be one to discuss Monday -in hindsight, we can probably cut it from a commit anyway so this shouldn't be a blocker. 

The impact, i think, will be measured in how many users we're supporting that we have on `1.7`, `1.8`, and `1.9` eventually, and easily being able to switch between those.

**Comment by jraddaoui** [comment]
I like the idea of supporting stable versions too. So far, I've never trusted anything behind the submodules commits in here and I use the Vagrant Boxes for that matter.

**Pull request merged**